README.md

```diff
@@ -62,66 +62,6 @@ Argo Workflows is an open source container-native workflow engine for orchestrat
 * DinD (docker-in-docker)
 * Script steps
 
-## Who uses Argo?
-As the Argo Community grows, we'd like to keep track of our users. Please send a PR with your organization name.
```

Comment:
Instead of removing this whole section, why not show a couple "featured" users and link to `USERS.md` from here under "See all users?"

Reply:
I added a link to the USERS file. "Featured" would be a subjective list, so I would like to avoid that.

Reply:
Fair point!